test/junit/scala/collection/immutable/LazyListTest.scala

```diff
@@ -444,8 +444,20 @@ class LazyListTest {
     LazyList(1).lazyAppendedAll({ count += 1; Seq(2)}).toList
     assertEquals(1, count)
   }
+
+  @Test
+  def lazyRangeAllowsMoreThanIntMaxValue(): Unit = {
+    val totalElements: Long = Int.MaxValue.toLong + 2L
+    val count: Long =
+      LazyList
+        .range(start = 0L, end = totalElements)
+        .foldLeft(0L) { case (acc, _) =>
+          acc + 1L
+        }
```

Comment:
even with inlining turned on, it's a lot to ask of a unit test. Maybe it's sufficient to test construction and head.

Reply:
Yeah I was wondering the same.
However, not sure if just asking for the head ensures there are more than `Int.MaxValue` elements.

So not sure if there is a smart way to test this without it being a CPU trap.

Reply:
Using a reverse `range` we don't need to compute a lot of elements.
Let me know what you think of the new test.
C.C. @NthPortal